src/components/bookList/BookList.tsx

```diff
@@ -1,52 +1,69 @@
 import { useEffect, useState } from 'react'
 
+import { AxiosResponse } from 'axios'
 import InfiniteScroll from 'react-infinite-scroll-component'
 
 import { Book } from '../../model/Book'
-import BookRequest from '../../model/BookRequest'
 import BookService from '../../services/BookService'
 import Card from '../card/Card'
 import './BookList.css'
+import Search from '../search/Search'
+import Where from '../../model/Where'
+import BookResponse from '../../model/BookResponse'
 
-const numberOfCard = 24
+interface BooksRequest {
+  PageNumber: number,
+  PageLength: number,
+  Where: Where[],
+}
+const initialItem = 6
 const BookList = () => {
-  const [ book, setBook ] = useState<Book[]>([])
+  const [ books, setBooks ] = useState<Book[]>([])
   const [ hasMore, setHasMore ] = useState(true)
-  const [ pageNubmer, setPageNumber ] = useState(1)
-
-  useEffect(() => {
-    getDataBooks()
-  }, [ pageNubmer ])
+  const [ pageNumber, setPageNumber ] = useState(1)
+  const [ searchValue, setSearchValue ] = useState('')
+  const [ currentSearch, setCurrentSearch ] = useState<string>(searchValue)
 
   const nexPage = () => {
-    setPageNumber( pageNubmer + 1)
+    setPageNumber(pageNumber => pageNumber + 1)
   }
 
-  const getDataBooks = async () => {
-    const bookRequest : BookRequest = {
-      PageNumber: pageNubmer,
-      PageLength: numberOfCard
-    }
-    try{
-      const response = await BookService.getAllBooks(bookRequest)
-      setHasMore(pageNubmer * numberOfCard <= response.data.TotalCount)
-      setBook([ ...book,...response.data.Items ])
-    } catch(error) {
-      console.error(error)
+  const dataBooks = () => {
```

Comment:
Not very intuitive name for this function, could you name it somehow to describe what is that function doing?

Reply:
Ok i fix this.